### DISCUSSION TITLE
cannot load js and css assets

### DISCUSSION THREAD

**Discussion by TheBlindHawk** [phalcon/cphalcon#16074] [Q&A] [answered]
## My repository
I just started on Phalcon, so the issue might be unrelated to Phalcon's version.  
I created a new Phalcon 5.0.0RC4 repository and am following the steps on the [phalcon 5 docs](https://docs.phalcon.io/5.0/en/assets#custom) to add assets to my index page.  
IndexController.php
```php
<?php

use Phalcon\Mvc\Controller;

class IndexController extends Controller
{
    public function indexAction()
    {
        $this->assets->addCss('css/global.css');
        $this->assets->addJs('js/global.js');
    }
}
```
index.phtml
```html
<html>
	<head>
        <?php $this->assets->outputCss(); ?>
	</head>
	<body >
        // a lot of stuff here
        <?php $this->assets->outputJs(); ?>
	</body>
</html>
```
Overall structure:
```none
.
└── blindhawk
    ├── app
    │   ├── controllers
    │   │   └──IndexController.php
    │   └── views
    │       └──Index
    │           └──index.phtml
    └── public
        ├── css
        │    └──global.css
        ├── img
        ├── index.php
        └── js
             └──global.js
```
## My index output
My index page on the inspector (Elements)
```html
<!-- I removed the rest of the page code -->
<link rel="stylesheet" type="text/css" href="/css/global.css">
<script type="application/javascript" src="/js/global.js"></script>
```
Each containing this (Sources)
```none
Exception: CssController handler class cannot be loaded
Exception: JsController handler class cannot be loaded
```
And no error output on the Console.
How can I fix this?

**Comment by niden** [comment, marked as answer]
There seems to be a problem with your rewrite rules (.htaccess). The app tries to interpret the paths for your assets as controller paths.

Try changing a bit your code to something like:

```php
$this->assets->addCss('random/global.css');
```
and see that you get in the output 

```
<link rel="stylesheet" type="text/css" href="/random/global.css">
```
If after that your message is `RandomController handler class cannot be loaded` then it is a rewrite rule issue for sure.

**Comment by TheBlindHawk** [comment]
Yes, the message was indeed ```RandomController handler class cannot be loaded```.
I couldn't find any ```.htaccess``` files in my repository though.
Am I supposed to create one? I couldn't find it on the docs, maybe I skipped it?

**Reply by niden** [reply]
Have a look here on how to set your web server.

https://docs.phalcon.io/5.0/en/webserver-setup

**Reply by TheBlindHawk** [reply]
Thanks! It seems I had indeed skipped that part.
